api/src/users/users.service.ts

```diff
@@ -1,44 +1,75 @@
 import { CryptoService } from '@douglasneuroinformatics/nestjs/modules';
 import { ConflictException, Injectable } from '@nestjs/common';
-import { InjectModel } from '@nestjs/mongoose';
-import { Model } from 'mongoose';
+import { type ConfirmEmailInfo, type User } from '@prisma/client';
 import type { SetOptional } from 'type-fest';
 
-import { CreateUserDto } from './dto/create-user.dto';
-import { User, type UserDocument } from './schemas/user.schema';
+import { InjectModel } from '@/core/decorators/inject-prisma-client.decorator';
+import type { Model } from '@/prisma/prisma.types';
+
+import type { CreateUserDto } from './zod/user.js';
 
 @Injectable()
 export class UsersService {
   constructor(
-    @InjectModel(User.name) private userModel: Model<User>,
+    @InjectModel('User') private readonly userModel: Model<"User">,
     private readonly cryptoService: CryptoService
-  ) {}
+  ) { }
 
   /** Insert a new user into the database */
-  async createUser({ email, password, ...rest }: CreateUserDto): Promise<Omit<UserDocument, 'hashedPassword'>> {
-    const exists = await this.userModel.exists({ email });
-    if (exists) {
-      throw new ConflictException(`User with provided email already exists: ${email}`);
+  async createUser({ email, firstName, lastName, password, ...rest }: CreateUserDto): Promise<Omit<User, 'hashedPassword'>> {
+    const user_exists = await this.findByEmail(email);
+    if (user_exists) {
+      throw new ConflictException(`User with the provided email already exists: ${email}`);
     }
     const hashedPassword = await this.cryptoService.hashPassword(password);
     const createdUser = await this.userModel.create({
-      email,
-      hashedPassword,
-      ...rest
+      data: {
+        email,
+        firstName,
+        hashedPassword,
+        lastName,
+        ...rest
+      }
     });
 
-    const returnedUser: SetOptional<UserDocument, 'hashedPassword'> = createdUser;
+    const returnedUser: SetOptional<User, 'hashedPassword'> = createdUser;
     delete returnedUser.hashedPassword;
     return returnedUser;
   }
 
   /** Return the user with the provided email, or null if no such user exists */
   findByEmail(email: string) {
```

Comment:
Here, you can use the shorthand object notation `{ email }` instead of `{ email: email }`